src/main/java/hudson/matrix/MatrixBuild.java

```diff
@@ -399,7 +399,15 @@ private void listUpAggregators(Collection<?> values) {
 
         public void post2(BuildListener listener) throws Exception {
             for (MatrixAggregator a : aggregators)
-                a.endBuild();
+                // all aggregations will be performed even any of them failed.
+                try {
+                    if (!a.endBuild()) {
+                        listener.error(String.format("Aggregation failed for %s", a.toString()));
+                    }
+                } catch (Exception e) {
+                    e.printStackTrace(listener.error(String.format("Aggregation failed for %s", a.toString())));
```

Comment:
Exclude trace for AbortException, it expected behaviour, but print AbortException reason.